src/fides/api/ctl/database/samples.py

```diff
@@ -0,0 +1,151 @@
+"""
+Sample data to use for seeding the database. This are loaded & parsed from the
+sample project at src/fides/data/sample_project.
+
+See load_samples() in seed.py for usage.
+"""
+from importlib.resources import files
+from typing import Dict, List, Optional, TextIO, Union
+
+import yaml
+from expandvars import expandvars  # type: ignore
+from fideslang.models import Dataset, Organization, Policy, System
+from fideslang.validation import FidesKey
+
+import fides.data.sample_project  # type: ignore
+from fides.api.ops.schemas.connection_configuration.connection_config import (
+    CreateConnectionConfigurationWithSecrets,
+)
+
+
+def load_sample_resources_from_project(
+    strict: bool = False,
+) -> Dict[str, List[Union[Dataset, Organization, Policy, System]]]:
+    """
+    Loads all the sample resource YAML files from the sample project by
+    traversing through the sample_resources/ folder, inspecting each file, and
+    parsing them into fideslang models.
+
+    Returns a dictionary, grouped by resource type (e.g. Dataset, System...)
+
+    NOTE: This only implements parsing for types we _currently_ expect to use in
+    the sample project. If new types are added to the project, this will quietly
+    ignore them unless strict=True is provided, which throws an error. We test
+    the "strict" version of this in the integration tests to ensure this code
+    doesn't fall out of date.
+    """
+    sample_resources_path = files(fides.data.sample_project).joinpath(
+        "sample_resources/"
+    )
+    sample_resources_dict: Dict[str, List] = {
+        "dataset": [],
+        "organization": [],
+        "policy": [],
+        "system": [],
+    }
+    for yaml_path in sample_resources_path.iterdir():
+        with yaml_path.open("r") as file:
+            # NOTE: in load_sample_connections, we use `expandvars` to expand
+            # ENV vars into string to allow injecting secrets, etc. We don't do
+            # that for the sample_resources/*, however, to stay consistent with
+            # the behaviour of `fides push`
+            yaml_dict = yaml.safe_load(file)
+
+            # Parse the contents of the YAML into the respective Pydantic models
+            if yaml_dict.get("dataset", []):
+                datasets = [Dataset.parse_obj(e) for e in yaml_dict.get("dataset")]
+                sample_resources_dict["dataset"].extend(datasets)
```

Comment:
Note: this is effectively re-implementing _some_ of the guts of the logic that powers `fides push` to read resource files and deserialize them into Pydantic models. I think it's reasonable to do like this but does create some duplication that could be tech debt in the future.

Reply:
I want to make sure I'm not missing something here, is there a reason we can't reuse that code?

for instance, in `parse.py` there is a simple example of how this is done. It does return a proper Taxonomy object as well instead of a dict of objects

Reply:
You're not missing anything - I think we probably could. I'm a little hesitant to trust some of the file-loading logic (there's a reason I pursued `importlib` here, it's much more reliable for loading resources from within the package itself) but I think the core YAML reading & parsing could be reused...

I'll give it a quick try now but might need an assist

Reply:
That worked! Pushed a new commit. The only side-effect of `parse.py` that I don't like are the `print()` statements in there, which don't mix & match well with the logger... but it's OK, small price to pay.